src/main/java/io/cdap/plugin/azure/face/extractor/AzureFaceExtractorConfig.java

```diff
@@ -0,0 +1,163 @@
+
+  @Name(EMOTION_SUBSCRIPTION_KEY)
+  @Description("The Azure emotion API subscription key.")
+  @Macro
+  private String emotionSubscriptionKey;
+
+  public AzureFaceExtractorConfig(String sourceFieldName, Boolean continueOnError,
+                String facesSubscriptionKey, String emotionSubscriptionKey) {
+    this.sourceFieldName = sourceFieldName;
+    this.continueOnError = continueOnError;
+    this.facesSubscriptionKey = facesSubscriptionKey;
+    this.emotionSubscriptionKey = emotionSubscriptionKey;
+  }
+
+  private AzureFaceExtractorConfig(Builder builder) {
+    sourceFieldName = builder.sourceFieldName;
+    continueOnError = builder.continueOnError;
+    facesSubscriptionKey = builder.facesSubscriptionKey;
+    emotionSubscriptionKey = builder.emotionSubscriptionKey;
+  }
+
+  public static Builder builder() {
+    return new Builder();
+  }
+
+  public static Builder builder(AzureFaceExtractorConfig copy) {
+    return builder()
+      .setSourceFieldName(copy.sourceFieldName)
+      .setContinueOnError(copy.continueOnError)
+      .setFacesSubscriptionKey(copy.facesSubscriptionKey)
+      .setEmotionSubscriptionKey(copy.emotionSubscriptionKey);
+  }
+
+  public String getSourceFieldName() {
+    return sourceFieldName;
+  }
+
+  public boolean getContinueOnError() {
+    return continueOnError;
+  }
+
+  public String getFacesSubscriptionKey() {
+    return facesSubscriptionKey;
+  }
+
+  public String getEmotionSubscriptionKey() {
+    return emotionSubscriptionKey;
+  }
+
+  public void validate(FailureCollector failureCollector, Schema inputSchema) {
+    if (inputSchema == null) {
+      failureCollector.addFailure("Could not get the input schema to validate.",
+                                  "Please check input schema.");
+      return;
+    }
+    if (inputSchema.getField(sourceFieldName) == null) {
+      failureCollector.addFailure(
+        String.format("Source field '%s' was not present in the input schema. Schema: '%s'.", sourceFieldName,
+                      inputSchema.toString()), "Provide Source Field that present in the input schema.")
+        .withConfigProperty(SOURCE_FIELD_NAME);
+      return;
+    }
+    if (!inputSchema.getField(sourceFieldName).getSchema().isSimpleOrNullableSimple()) {
+      failureCollector.addFailure(
+        String.format("Source field must be a simple type but was type: '%s'.",
+                      inputSchema.getField(sourceFieldName).getSchema().getDisplayName()),
+        "Provide Source Field with simple type.")
+        .withConfigProperty(SOURCE_FIELD_NAME);
+    }
+    Schema.Type fieldType = inputSchema.getField(sourceFieldName).getSchema().isNullable()
+      ? inputSchema.getField(sourceFieldName).getSchema().getNonNullable().getType()
+      : inputSchema.getField(sourceFieldName).getSchema().getType();
+
+    if(fieldType != Schema.Type.BYTES) {
+      failureCollector.addFailure(
+        String.format("Input field must be bytes but was: '%s'.",
```

Comment:
error - `Source field '%s' is of unexpected type '%s'`
corrective action - `Ensure it is of type '%s'`

Reply:
Done.